### PR TITLE
Bugfix/turned into createSkinComponet

### PR DESCRIPTION
**Pull request by vik753**

### CODE REVIEW

File: epam-promo/components/overlays/NotificationCard.tsx [hunk 1]
```diff
@@ -1,21 +1,25 @@
-import { withMods } from '@epam/uui-core';
+import { createSkinComponent, devLogger } from '@epam/uui-core';
 import * as uui from '@epam/uui';
 import { EpamPrimaryColor } from '../types';
 
 interface NotificationCardMods {
     /**
-     * Defines component color.
```
Comment (siarhei-epam) [inline epam-promo/components/overlays/NotificationCard.tsx:7]: Why did you delete the point here when you add it in other descriptions?

File: uui/components/overlays/Tooltip.tsx [hunk 2]
```diff
@@ -1,16 +1,18 @@
 import { withMods } from '@epam/uui-core';
-import { Tooltip as uuiTooltip, TooltipProps as UuiTooltipProps } from '@epam/uui-components';
+import * as UuiComponents from '@epam/uui-components';
```
Comment (siarhei-epam) [inline uui/components/overlays/Tooltip.tsx:2]:
```suggestion
import * as uuiComponents from '@epam/uui-components';
```
use small letters for namespaces

File: uui/components/typography/Text.tsx [hunk 3]
```diff
@@ -1,23 +1,42 @@
 import { withMods } from '@epam/uui-core';
 import { getTextClasses, TextSettings } from '../../helpers';
-import { Text as uuiText, TextProps as UuiTextProps } from '@epam/uui-components';
+import * as UuiComponents from '@epam/uui-components';
```
Comment (siarhei-epam) [inline uui/components/typography/Text.tsx:3]:
```suggestion
import * as uuiComponents from '@epam/uui-components';
```
use small letters for namespaces

File: uui/components/datePickers/RangeDatePicker.tsx [hunk 4]
```diff
@@ -15,9 +15,16 @@ import { i18n } from '../../i18n';
 import css from './RangeDatePicker.module.scss';
 
 export interface RangeDatePickerProps extends BaseRangeDatePickerProps, SizeMod {
+    /**
+     * This callback returns a placeholder depending on the parameters passed.
+     * @param  { from | to }  type
```
Comment (AlekseyManetov) [inline uui/components/datePickers/RangeDatePicker.tsx:20]: Please rework it based on our guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#callbacks
This is the case of Pure function callback

File: uui/components/dnd/DropMarker.tsx [hunk 5]
```diff
@@ -3,7 +3,10 @@ import cx from 'classnames';
 import { DndActorRenderParams, IHasCX } from '@epam/uui-core';
 import css from './DropMarker.module.scss';
 
-export interface DropMarkerProps extends DndActorRenderParams, IHasCX {
+interface DropMarkerProps extends DndActorRenderParams, IHasCX {
+    /*
+    * Defines is a Blocker enable.
```
Comment (AlekseyManetov) [inline uui/components/dnd/DropMarker.tsx:8]: Please rework it based on our guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#scalars
For example:
```
/** Pass 'true' to enable Blocker
 * If false, nothing will be rendered
 */
```

File: uui/components/fileUpload/DropSpot.tsx [hunk 6]
```diff
@@ -9,10 +9,25 @@ import { i18n } from '../../i18n';
 import css from './DropSpot.module.scss';
 
 export interface DropSpotProps extends IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
+    /*
+    * CSS class(es) to put on component's root. See {@link https://github.com/JedWatson/classnames#usage} for details
+    */
     cx?: string;
```
Comment (AlekseyManetov) [inline uui/components/fileUpload/DropSpot.tsx:15]: Extend interface from `IHasCX` instead of declare cx directly in interface

File: uui/components/fileUpload/DropSpot.tsx [hunk 7]
```diff
@@ -9,10 +9,25 @@ import { i18n } from '../../i18n';
...
+    /*
+    * Defines the infoText.
+    */
     infoText?: string;
+    /*
+    * Callback to processing upload files
+    */
     onUploadFiles(files: File[]): any;
```
Comment (AlekseyManetov) [inline uui/components/fileUpload/DropSpot.tsx:23]: Rework by guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#api-design-and-documentation-guidelines

File: uui/components/fileUpload/DropSpot.tsx [hunk 8]
```diff
@@ -9,10 +9,25 @@ import { i18n } from '../../i18n';
 import css from './DropSpot.module.scss';
 
 export interface DropSpotProps extends IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
+    /*
+    * CSS class(es) to put on component's root. See {@link https://github.com/JedWatson/classnames#usage} for details
+    */
     cx?: string;
+    /*
+    * Defines the infoText.
```
Comment (AlekseyManetov) [inline uui/components/fileUpload/DropSpot.tsx:17]: Add more clear description what is the infoText - where it will be rendered and for which cases?

File: uui/components/fileUpload/FileCard.tsx [hunk 9]
```diff
@@ -13,12 +13,25 @@ import { ReactComponent as RemoveIcon } from '@epam/assets/icons/common/navigati
 import { ReactComponent as ErrorIcon } from '../../icons/notification-error-fill-10.svg';
 
 export interface FileCardItem extends Partial<File & FileUploadResponse> {
+    /*
+    * Defines upload progress.
```
Comment (AlekseyManetov) [inline uui/components/fileUpload/FileCard.tsx:17]: Also add - 'It should be number value in range from 0 to 100', where 100 means that file fully loaded.

File: uui/components/layout/Accordion.tsx [hunk 10]
```diff
@@ -3,12 +3,19 @@ import * as uuiComponents from '@epam/uui-components';
 import { systemIcons } from '../../icons/icons';
 import css from './Accordion.module.scss';
 
-export interface AccordionMods {
-    /** @default 'block' */
+interface AccordionMods {
+    /**
+     * Defines component positioning mode.
```
Comment (AlekseyManetov) [inline uui/components/layout/Accordion.tsx:8]: let's change to -     * Defines component displaying mode.

File: uui/components/layout/Accordion.tsx [hunk 11]
```diff
@@ -3,12 +3,19 @@ import * as uuiComponents from '@epam/uui-components';
...
+     * @default 'block'
+     */
     mode?: 'block' | 'inline';
+    /*
+    * Defines component padding.
```
Comment (AlekseyManetov) [inline uui/components/layout/Accordion.tsx:13]: clarify which padding it defines - vertical or horizontal?

File: uui/components/layout/FlexItems/FlexRow.tsx [hunk 12]
```diff
@@ -3,18 +3,49 @@ import { FlexRow as uuiFlexRow } from '@epam/uui-components';
...
+    /*
+    * Defines row gap.
+    */
     rowGap?: number | '6' | '12' | '18' | '24' | '36';
+    /*
+    * Defines row spacing.
+    */
     spacing?: '6' | '12' | '18';
+    /*
+    * Defines is a top shadow showing.
```
Comment (AlekseyManetov) [inline uui/components/layout/FlexItems/FlexRow.tsx:35]: Rework according boolean guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#api-design-and-documentation-guidelines

File: uui/components/layout/FlexItems/Panel.tsx [hunk 13]
```diff
@@ -3,11 +3,21 @@ import { VPanel } from '@epam/uui-components';
 import css from './Panel.module.scss';
 
 interface PanelMods {
+    /*
```
Comment (AlekseyManetov) [inline uui/components/layout/FlexItems/Panel.tsx:6]: Rework according boolean guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#api-design-and-documentation-guidelines

File: uui/components/navigation/MainMenu/MainMenuButton.tsx [hunk 14]
```diff
@@ -3,7 +3,10 @@ import { Button, ButtonProps } from '@epam/uui-components';
 import { IAdaptiveItem, withMods } from '@epam/uui-core';
 import { ReactComponent as SvgTriangle } from '../../../icons/chevron-down-24.svg';
 
-export interface MainMenuButtonMods {
+interface MainMenuButtonMods {
+    /*
+    * Defines component type.
```
Comment (AlekseyManetov) [inline uui/components/navigation/MainMenu/MainMenuButton.tsx:8]: Need to clarify what is the difference between 'primary' and 'secondary'

File: uui/components/overlays/Alert.tsx [hunk 15]
```diff
@@ -11,15 +11,24 @@ import { ReactComponent as CrossIcon } from '../../icons/navigation-close-24.svg
 import css from './Alert.module.scss';
 
 type notificationAction = {
+    /*
+    * Defines notification name.
```
Comment (AlekseyManetov) [inline uui/components/overlays/Alert.tsx:15]: It's not an notification name, it's name of the action which will be action caption

File: uui/components/overlays/Alert.tsx [hunk 16]
```diff
@@ -11,15 +11,24 @@ import { ReactComponent as CrossIcon } from '../../icons/navigation-close-24.svg
...
+    */
     name: string;
+    /*
+    * Callback to processing of the notification action.
```
Comment (AlekseyManetov) [inline uui/components/overlays/Alert.tsx:19]: /** Called when action is clicked */

File: uui/components/overlays/NotificationCard.tsx [hunk 17]
```diff
@@ -11,19 +11,26 @@ import { ReactComponent as HintIcon } from '../../icons/notification-help-fill-2
 import { ReactComponent as CrossIcon } from '../../icons/snackbar/cross.svg';
 import css from './NotificationCard.module.scss';
 
-interface NotificationAction extends IHasRawProps<React.ButtonHTMLAttributes<HTMLButtonElement>> {
+export interface NotificationAction extends IHasRawProps<React.ButtonHTMLAttributes<HTMLButtonElement>> {
+    /** NotificationAction name */
     name: string;
+    /** Notification action */
```
Comment (AlekseyManetov) [inline uui/components/overlays/NotificationCard.tsx:17]: /** Called when action is clicked */

File: uui/components/overlays/NotificationCard.tsx [hunk 18]
```diff
@@ -11,19 +11,26 @@ import { ReactComponent as HintIcon } from '../../icons/notification-help-fill-2
...
     action: () => void;
 }
 
-export interface DefaultNotificationProps extends INotification, IHasChildren, IHasCX, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
+export type NotificationCardCoreProps = INotification & IHasChildren & IHasCX & IHasRawProps<React.HTMLAttributes<HTMLDivElement>> & {
+    /** Array of Notification actions */
```
Comment (AlekseyManetov) [inline uui/components/overlays/NotificationCard.tsx:22]: Add - 'If provided will be displayed as LinkButtons in the end of notification'

File: uui/components/typography/Text.tsx [hunk 19]
```diff
@@ -1,23 +1,42 @@
...
 
 type TextColor = 'info' | 'warning' | 'error' | 'success' | 'brand' | 'primary' | 'secondary' | 'disabled' | 'white';
 type TextSize = 'none' | '18' | '24' | '30' | '36' | '42' | '48';
 type TextFontStyle = 'normal' | 'italic';
 type TextFontWeight = '200' | '300' | '400' | '600' | '700' | '900';
 
-interface TextMods extends TextSettings {
+interface TextMods {
+    /**
+     * Represents the color of a text.
```
Comment (AlekseyManetov) [inline uui/components/typography/Text.tsx:13]: Change to - 'Defines text color'

File: uui/components/typography/Text.tsx [hunk 20]
```diff
@@ -1,23 +1,42 @@
...
+     * @default '400'
+     */
     fontWeight?: TextFontWeight;
+    /**
+     * Determines the style of the text font.
+     * @default 'normal'
+     */
     fontStyle?: TextFontStyle;
+    /**
+     * Represents the size of a text.
```
Comment (AlekseyManetov) [inline uui/components/typography/Text.tsx:31]: Change to - 'Defines text size'

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: epam-promo/components/layout/IconContainer.tsx [hunk 21]
```diff
@@ -7,7 +7,7 @@ type IconContainerColors = 'blue' | 'green' | 'amber' | 'red' | 'cyan' | 'orange
 
 interface IconContainerMods {
     /**
-     *  IconContainer color.
+     *  Defines component size.
```
[the remark this answers is not in the document]
Comment (vik753) [reply epam-promo/components/layout/IconContainer.tsx:10]: fixed

File: epam-promo/components/overlays/NotificationCard.tsx (hunk 1 above)
Comment (vik753) [reply epam-promo/components/overlays/NotificationCard.tsx:7]: fixed

File: uui/components/datePickers/RangeDatePicker.tsx (hunk 4 above)
Comment (vik753) [reply uui/components/datePickers/RangeDatePicker.tsx:20]: fixed

File: uui/components/dnd/DropMarker.tsx (hunk 5 above)
Comment (vik753) [reply uui/components/dnd/DropMarker.tsx:8]: fixed

File: uui/components/fileUpload/DropSpot.tsx (hunk 6 above)
Comment (vik753) [reply uui/components/fileUpload/DropSpot.tsx:15]: fixed

File: uui/components/fileUpload/DropSpot.tsx (hunk 7 above)
Comment (vik753) [reply uui/components/fileUpload/DropSpot.tsx:23]: fixed

File: uui/components/fileUpload/DropSpot.tsx (hunk 8 above)
Comment (vik753) [reply uui/components/fileUpload/DropSpot.tsx:17]: fixed

File: uui/components/layout/FlexItems/FlexRow.tsx (hunk 12 above)
Comment (vik753) [reply uui/components/layout/FlexItems/FlexRow.tsx:35]: fixed

File: uui/components/layout/FlexItems/Panel.tsx (hunk 13 above)
Comment (vik753) [reply uui/components/layout/FlexItems/Panel.tsx:6]: fixed

File: uui/components/navigation/MainMenu/MainMenuButton.tsx (hunk 14 above)
Comment (vik753) [reply uui/components/navigation/MainMenu/MainMenuButton.tsx:8]: fixed

File: uui/components/overlays/Alert.tsx (hunk 15 above)
Comment (vik753) [reply uui/components/overlays/Alert.tsx:15]: fixed

File: uui/components/overlays/Alert.tsx (hunk 16 above)
Comment (vik753) [reply uui/components/overlays/Alert.tsx:19]: fixed

File: uui/components/overlays/NotificationCard.tsx (hunk 17 above)
Comment (vik753) [reply uui/components/overlays/NotificationCard.tsx:17]: fixed

File: uui/components/overlays/NotificationCard.tsx (hunk 18 above)
Comment (vik753) [reply uui/components/overlays/NotificationCard.tsx:22]: fixed

File: uui/components/typography/Text.tsx (hunk 19 above)
Comment (vik753) [reply uui/components/typography/Text.tsx:13]: fixed

File: uui/components/typography/Text.tsx (hunk 20 above)
Comment (vik753) [reply uui/components/typography/Text.tsx:31]: fixed

File: uui/components/datePickers/DatePickerBody.tsx [hunk 22]
```diff
@@ -2,17 +2,21 @@ import * as React from 'react';
...
 
 dayjs.extend(updateLocale);
 
 export interface DatePickerBodyProps extends DatePickerBodyBaseProps<string> {
+    /*
+    * Array of classes for styling the Day.
+    */
     getDayCX?(day: Dayjs): string[];
+    /*
+    * Boolean flag that showing is this Day a Holiday.
```
Comment (AlekseyManetov) [inline uui/components/datePickers/DatePickerBody.tsx:18]: Please rework it based on our guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#callbacks
This is the case of Pure function callback

File: uui/components/datePickers/DatePickerBody.tsx [hunk 23]
```diff
@@ -2,17 +2,21 @@ import * as React from 'react';
...
+import { MonthSelection, YearSelection, DatePickerBodyBase, DatePickerBodyBaseProps, valueFormat, i18n } from '@epam/uui-components';
 import { DatePickerHeader } from './DatePickerHeader';
 import { Calendar } from './Calendar';
 import css from './DatePickerBody.module.scss';
 
 dayjs.extend(updateLocale);
 
 export interface DatePickerBodyProps extends DatePickerBodyBaseProps<string> {
+    /*
+    * Array of classes for styling the Day.
```
Comment (AlekseyManetov) [inline uui/components/datePickers/DatePickerBody.tsx:14]: Please rework it based on our guidelines - https://github.com/epam/UUI/wiki/Documentation-Guidelines#callbacks
This is the case of Pure function callback

File: uui/components/datePickers/RangeDatePicker.tsx (hunk 4 above)
Comment (AlekseyManetov) [reply uui/components/datePickers/RangeDatePicker.tsx:20]: still doesn't match the guidelines

File: uui/components/datePickers/DatePickerHeader.tsx [hunk 24]
```diff
@@ -20,7 +20,13 @@ export const uuiHeader = {
 };
 
 export interface DatePickerHeaderProps extends IEditable<PickerBodyValue<string>>, IHasCX {
+    /*
+    * Represents a navigation icon on the left side.
```
Comment (AlekseyManetov) [inline uui/components/datePickers/DatePickerHeader.tsx:24]: Let's change to  - 
```
/** Navigation icon for the left navigation icon in header. 
 * Usually it has a default implementation in skins, so providing this is only necessary if you want to replace the default icon
*/
```

File: uui/components/datePickers/DatePickerHeader.tsx [hunk 25]
```diff
@@ -20,7 +20,13 @@ export const uuiHeader = {
...
+    */
     navIconLeft?: Icon;
+    /*
+    * Represents a navigation icon on the right side.
```
Comment (AlekseyManetov) [inline uui/components/datePickers/DatePickerHeader.tsx:28]: The same as for previous prop

File: uui/components/dnd/DropMarker.tsx [hunk 26]
```diff
@@ -3,7 +3,11 @@ import cx from 'classnames';
 import { DndActorRenderParams, IHasCX } from '@epam/uui-core';
 import css from './DropMarker.module.scss';
 
-export interface DropMarkerProps extends DndActorRenderParams, IHasCX {
+interface DropMarkerProps extends DndActorRenderParams, IHasCX {
+    /**
+     * Pass 'true' to enable Blocker
+     * If false, nothing will be rendered
```
Comment (AlekseyManetov) [inline uui/components/dnd/DropMarker.tsx:9]: Let's remove 'If false, nothing will be rendered'. Sorry, it was my mistake in previous suggestion.

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui/components/datePickers/DatePickerBody.tsx (hunk 22 above)
Comment (vik753) [reply uui/components/datePickers/DatePickerBody.tsx:18]: fixed

File: uui/components/datePickers/DatePickerBody.tsx (hunk 23 above)
Comment (vik753) [reply uui/components/datePickers/DatePickerBody.tsx:14]: fixed
